worker/workercontroller.go

```diff
@@ -32,41 +29,38 @@ func (c *Controller) List(filterStrings ...string) ([]Worker, error) {
 		return nil, err
 	}
 	var workers []Worker
-	c.workers.Range(func(k, v interface{}) bool {
-		w := v.(Worker)
+	for _, w := range c.workers {
 		if filter.Match(adaptWorker(w)) {
 			workers = append(workers, w)
 		}
-		return true
-	})
+	}
 	return workers, nil
 }
 
 // GetDefault returns the default local worker
 func (c *Controller) GetDefault() (Worker, error) {
-	if c.defaultID == "" {
+	if len(c.workers) == 0 {
 		return nil, errors.Errorf("no default worker")
 	}
-	return c.Get(c.defaultID)
+	return c.workers[0], nil
 }
 
 func (c *Controller) Get(id string) (Worker, error) {
-	v, ok := c.workers.Load(id)
-	if !ok {
-		return nil, errors.Errorf("worker %s not found", id)
+	for _, w := range c.workers {
```

Comment:
I guess this isn't used much atm. Otherwise, would be good to have something faster than full scan.